src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.Parsers.cs

```diff
@@ -67,24 +79,35 @@ internal static bool DetermineIfNewLineOptionIsSet(string value, NewLineOption o
                     .Any();
         }
 
-
         private static NewLineOption? ConvertToNewLineOption(string value)
+            => s_newLineOptionsEditorConfigMap.GetValueOrDefault(value);
+        private static string GetNewLineOptionEditorConfigString(OptionSet optionSet)
         {
-            switch (value.Trim())
+            List<string> editorConfigStringBuilderOpt = null;
+            foreach (var kvp in NewLineOptionsMap)
+            {
+                var value = optionSet.GetOption(kvp.Key);
+                if (value)
+                {
+                    editorConfigStringBuilderOpt = editorConfigStringBuilderOpt ?? new List<string>(NewLineOptionsMap.Count);
```

Comment:
note: this is not a heavily allocation codepath.  consider just simplifying by always initializing htis, and then just checking for empty below.